drivers/virtualbox/virtualbox.go

```diff
@@ -363,6 +363,14 @@ func (d *Driver) Create() error {
 		return err
 	}
 
+	cmd, err = d.GetSSHCommand(fmt.Sprintf("sudo hostname %s", d.MachineName))
```

Comment:
Does that persist on boot2docker? Perhaps we don't bother setting the hostname on boot2docker... I think the big win from this pull request is the correct name showing up in your cloud provider's admin (e.g. list of instances on EC2), not the hostname when you log in.